[BasicAA] Treat ExtractValue(Argument) similar to Argument in relation to function-local objects.

davemgreen opened the pull request:
This is a much smaller, technically orthogonal patch similar to #134505. It
states that a extractvalue(Argument) can be treated like an Argument for alias
analysis, where the extractelement acts like a phi / copy. No inttoptr here. I'm
hoping for this operations this is perfectly sensible, but it is quite a
complex area.

nikic reviewed (approved):
Yes, this case is fine. LGTM

nunoplopes reviewed (approved):
LGTM too.

davemgreen committed:
[BasicAA] Treat ExtractValue(Argument) similar to Argument in relation to function-local objects.

This is a much smaller, technically orthoganal patch similar to #134505. It
states that a extractvalue(Argument) can be treated like an Argument for alias
analysis, where the extractelement acts like a phi / copy. No inttotr here. I'm
hoping for this operations this is perfectly sensible, but it is quite a
complex area.

davemgreen commented:
Thanks.
